src/__swaps__/screens/Swap/resources/search/search.ts

```diff
@@ -0,0 +1,126 @@
+
+type TokenSearchQueryKey = ReturnType<typeof tokenSearchQueryKey>;
+
+// ///////////////////////////////////////////////
+// Query Function
+
+async function tokenSearchQueryFunction({
+  queryKey: [{ chainId, fromChainId, keys, list, threshold, query }],
+}: QueryFunctionArgs<typeof tokenSearchQueryKey>) {
+  const queryParams: {
+    keys?: string;
+    list: TokenSearchListId;
+    threshold?: TokenSearchThreshold;
+    query?: string;
+    fromChainId?: number | string;
+  } = {
+    keys: keys?.join(','),
+    list,
+    threshold,
+    query,
+    fromChainId: fromChainId,
+  };
+
+  const isAddressSearch = query && isAddress(query);
+
+  if (isAddressSearch) {
+    queryParams.keys = `networks.${chainId}.address`;
+  }
+
+  const url = `${chainId ? `/${chainId}` : ''}/?${qs.stringify(queryParams)}`;
+  const isSearchingVerifiedAssets = queryParams.list === 'verifiedAssets';
+
+  try {
+    if (isAddressSearch && isSearchingVerifiedAssets) {
+      const tokenSearch = await tokenSearchHttp.get<{ data: SearchAsset[] }>(url);
+
+      if (tokenSearch && tokenSearch.data.data.length > 0) {
+        return parseTokenSearch(tokenSearch.data.data, chainId);
+      }
+
+      const allVerifiedTokens = await tokenSearchHttp.get<{ data: SearchAsset[] }>(ALL_VERIFIED_TOKENS_PARAM);
+
+      const addressQuery = query.trim().toLowerCase();
+      const addressMatchesOnOtherChains = allVerifiedTokens.data.data.filter(a =>
+        Object.values(a.networks).some(n => n?.address === addressQuery)
+      );
+
+      return parseTokenSearch(addressMatchesOnOtherChains);
+    } else {
+      const tokenSearch = await tokenSearchHttp.get<{ data: SearchAsset[] }>(url);
+      return parseTokenSearch(tokenSearch.data.data, chainId);
+    }
+  } catch (e) {
+    logger.error(new RainbowError('Token search failed'), { url });
+    return [];
+  }
+}
+
+export type TokenSearchResult = QueryFunctionResult<typeof tokenSearchQueryFunction>;
+
+// ///////////////////////////////////////////////
+// Query Fetcher
+
+export async function fetchTokenSearch(
+  { chainId, fromChainId, keys, list, threshold, query }: TokenSearchArgs,
+  config: QueryConfigWithSelect<TokenSearchResult, Error, TokenSearchResult, TokenSearchQueryKey> = {}
+) {
+  return await queryClient.fetchQuery(
+    tokenSearchQueryKey({ chainId, fromChainId, keys, list, threshold, query }),
+    tokenSearchQueryFunction,
+    config
+  );
+}
+
+// ///////////////////////////////////////////////
+// Query Hook
+
+export function useTokenSearch(
+  { chainId, fromChainId, keys, list, threshold, query }: TokenSearchArgs,
+  config: QueryConfigWithSelect<TokenSearchResult, Error, TokenSearchResult, TokenSearchQueryKey> = {}
+) {
+  return useQuery(
+    tokenSearchQueryKey({ chainId, fromChainId, keys, list, threshold, query }),
+    tokenSearchQueryFunction,
+    config ? { ...config, keepPreviousData: true } : { keepPreviousData: true }
```

Comment:
```suggestion
    { ...config, keepPreviousData: true }
```
this can be simplified to the above since we default config to `{}`